_data/rules.yml

```diff
@@ -223,6 +223,10 @@ groups:
                 description: '{{ $labels.instance }} interface {{ $labels.device }} has encountered {{ printf "%.0f" $value }} transmit errors in the last five minutes.'
                 query: 'increase(node_network_transmit_errs_total[5m]) > 0'
                 severity: warning
+              - name: Host filesystem readonly
+                description: '{{ $labels.instance }} partition {{ $labels.device }} has become readonly.'
+                query: 'node_filesystem_readonly{fstype!~"(tmpfs|proc|aufs|rpc_pipefs)"} == 1'
```

Comment:
This query doesn't reflect that filesystem **become** read-only. It only states that filesystem **is** readonly. This means it will fire even when filesystem is read-only since the beginning as it doesn't look for past data.

Reply:
Possibly something like:
```
node_filesystem_readonly > 0 AND changes(node_filesystem_readonly[5m]) > 0
```
would solve this.
Above query is not tested though.